From 53334366136448738dc76e4cc25fa453152ffe7f Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Tue, 18 Dec 2012 17:59:20 +0100
Subject: [PATCH] Enable SSSD on client install

authconfig component changed its behavior. It no longer starts and
enables SSSD daemon when --enablesssd and --enablesssdauth options
are used. It only enables the PAM module and adds SSSD to nsswitch.

Enable SSSD on new client/server installs manually. Also make sure
that we stop&disable SSSD when we delete the configuration.

https://fedorahosted.org/freeipa/ticket/3307
---
 ipa-client/ipa-install/ipa-client-install | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/ipa-client/ipa-install/ipa-client-install b/ipa-client/ipa-install/ipa-client-install
index 1c68814d5677284796af2f02f9f751de9bba52cc..d548c891da9630f200a69f3b9f81f0cb45e5b65e 100755
--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -500,6 +500,18 @@ def uninstall(options, env):
         root_logger.info("Redundant SSSD configuration file " +
             "/etc/sssd/sssd.conf was moved to /etc/sssd/sssd.conf.deleted")
 
+        sssd = ipaservices.service('sssd')
+        try:
+            sssd.stop()
+        except CalledProcessError:
+            root_logger.warning("SSSD service could not be stopped")
+
+        try:
+            sssd.disable()
+        except CalledProcessError, e:
+            root_logger.warning(
+                "Failed to disable automatic startup of the SSSD daemon: %s", e)
+
     if fstore.has_files():
         root_logger.info("Restoring client configuration files")
         fstore.restore_all_files()
@@ -2188,6 +2200,12 @@ def install(options, env, fstore, statestore):
             except CalledProcessError:
                 root_logger.warning("SSSD service restart was unsuccessful.")
 
+            try:
+                sssd.enable()
+            except CalledProcessError, e:
+                root_logger.warning(
+                    "Failed to enable automatic startup of the SSSD daemon: %s", e)
+
         if not options.sssd:
             #Modify pam to add pam_krb5 only when sssd is not in use
             auth_config.reset()
-- 
1.7.11.7

